schneiderWallbox/integrationpluginschneiderwallbox.cpp

```diff
@@ -0,0 +1,546 @@
+*
+* Copyright 2013 - 2020, nymea GmbH
+* Contact: contact@nymea.io
+*
+* This file is part of nymea.
+* This project including source code and documentation is protected by
+* copyright law, and remains the property of nymea GmbH. All rights, including
+* reproduction, publication, editing and translation, are reserved. The use of
+* this project is subject to the terms of a license agreement to be concluded
+* with nymea GmbH in accordance with the terms of use of nymea GmbH, available
+* under https://nymea.io/license
+*
+* GNU Lesser General Public License Usage
+* Alternatively, this project may be redistributed and/or modified under the
+* terms of the GNU Lesser General Public License as published by the Free
+* Software Foundation; version 3. This project is distributed in the hope that
+* it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+* warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this project. If not, see <https://www.gnu.org/licenses/>.
+*
+* For any further details and any questions please contact us under
+* contact@nymea.io or see our FAQ/Licensing Information on
+* https://nymea.io/license/faq
+*
+* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#include "integrationpluginschneiderwallbox.h"
+#include "plugininfo.h"
+
+#include <network/networkdevicediscovery.h>
+#include <types/param.h>
+#include <plugintimer.h>
+
+#include <QDebug>
+#include <QStringList>
+#include <QJsonDocument>
+#include <QNetworkInterface>
+
+
+IntegrationPluginSchneiderWallbox::IntegrationPluginSchneiderWallbox()
+{
+
+}
+
+void IntegrationPluginSchneiderWallbox::discoverThings(ThingDiscoveryInfo *info)
+{
+    if (!hardwareManager()->networkDeviceDiscovery()->available()) {
+        qCWarning(dcSchneiderElectric()) << "Failed to discover network devices. The network device discovery is not available.";
+        info->finish(Thing::ThingErrorHardwareNotAvailable, QT_TR_NOOP("The discovery is not available. Please enter the IP address manually."));
+        return;
+    }
+
+    qCDebug(dcSchneiderElectric()) << "Starting network discovery...";
+    NetworkDeviceDiscoveryReply *discoveryReply = hardwareManager()->networkDeviceDiscovery()->discover();
+    connect(discoveryReply, &NetworkDeviceDiscoveryReply::finished, info, [=](){
+        qCDebug(dcSchneiderElectric()) << "Discovery finished. Found" << discoveryReply->networkDeviceInfos().count() << "devices";
+        foreach (const NetworkDeviceInfo &networkDeviceInfo, discoveryReply->networkDeviceInfos()) {
+            qCDebug(dcSchneiderElectric()) << networkDeviceInfo;
+
+            if (!networkDeviceInfo.macAddressManufacturer().contains("TELEMECANIQUE", Qt::CaseSensitivity::CaseInsensitive)) {
```

Comment:
This will cause all devices made by TELEMECANIQUE to appear as such a wallbox. Sounds like it's not enough to filter properly.

Also, are you sure that all models will have mac addresses registered by TELEMATIQUE? Reading up on the history of the TELEMATIQUE - Schneider merger, my personal guess would be that they will use up macs registered by telematique and then continue with macs on their own.

This seems like it needs some more thoughts unless you can confirm that the assumptions made will hold (all macs by TELEMATIQUE will be used for the wallbox and all wallboxes will only use macs by TELEMATIQUE)